### PR TITLE
auth: redo implementation with a cleaner and simpler API surface

### PR DESCRIPTION
**Pull request by msfjarvis**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring

## :scroll: Description
Coalesce the logic into a single file and make some improvements to security
by explicitly handling all return values.

## :bulb: Motivation and Context
This was due for a while and I learned a few things when I worked on a similar
feature for WireGuard which I wanted to bring here.

## :green_heart: How did you test it?
Verify biometric auth behaves as before.

## :pencil: Checklist

- [x] I ran `./gradlew spotlessApply` before submitting the PR
- [x] I reviewed submitted code

### CODE REVIEW

File: app/src/main/java/com/zeapo/pwdstore/utils/BiometricAuthenticator.kt [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                callback(Result.Success(result.cryptoObject))
+            }
+        }
+        val biometricPrompt = BiometricPrompt(activity, { handler.post(it) }, authCallback)
+        val promptInfo = BiometricPrompt.PromptInfo.Builder()
+                .setTitle(activity.getString(dialogTitleRes))
+                .setDeviceCredentialAllowed(true)
+                .build()
+        if (BiometricManager.from(activity).canAuthenticate() == BiometricManager.BIOMETRIC_SUCCESS ||
+                activity.getSystemService<KeyguardManager>()!!.isDeviceSecure) {
```
Comment (FabianHenneke) [inline app/src/main/java/com/zeapo/pwdstore/utils/BiometricAuthenticator.kt:68]: Maybe use ?. here since we have a dedicated hardware error anyway?

Comment (msfjarvis) [reply app/src/main/java/com/zeapo/pwdstore/utils/BiometricAuthenticator.kt]: Sure, made the change.

**Comment by Skrilltrax** [comment]
Does this still not lock the app when it goes to background ?

**Comment by msfjarvis** [comment]
> Does this still not lock the app when it goes to background ?

No I don't bother with resume behavior.

**Comment by Skrilltrax** [comment]
Any plan to include it in future ?

**Comment by msfjarvis** [comment]
> Any plan to include it in future ?

I don't personally care for it. If you wanna add it I'll happily review the PR.

**Comment by Skrilltrax** [comment]
Sure

**Pull request merged**